core/primitives/src/errors.rs

```diff
@@ -202,6 +202,13 @@ pub enum ActionsValidationError {
     DelegateActionCantContainNestedOne,
     /// There should be the only one DelegateAction
     DelegateActionMustBeOnlyOne,
+    /// The transaction includes a feature that the current protocol version
+    /// does not support.
+    ///
+    /// Note: we stringify the protocol feature name instead of using
+    /// `ProtocolFeature` here because we don't want to leak the internals of
+    /// that type into observable borsh serialization.
+    UnsupportedProtocolFeature { protocol_feature: String },
```

Comment:
consider also adding a protocol_version information to this error

Reply:
makes sense, done